src/reducers/MessageCenter.ts

```diff
@@ -0,0 +1,101 @@
+import { MessageCenterActionType } from '../actions';
+import { MessageType } from '../types/MessageCenter';
+import { MessageCenterState } from '../store/Store';
+
+const INITIAL_STATE: MessageCenterState = {
+  nextId: 0,
+  groups: [
+    {
+      id: 'default',
+      title: 'Default',
+      messages: []
+    }
+  ],
+  hidden: true,
+  expanded: false,
+  expandedGroupId: 'default'
+};
+
+const mergeToState = (prevState, toMerge) => {
+  return Object.assign({}, prevState, toMerge);
```

Comment:
Personally, I prefer the ES6 spread operator because the syntax seem cleaner (just my preference though). 
We could add `tassign` if we want `Object.assign` to be typed.

Reply:
Used the spread operator in latest commit.